### PR TITLE
Fixes #31535 - check for satellite and capsule packages

### PR DESCRIPTION
**Pull request by upadhyeammit**

### CODE REVIEW

File: lib/foreman_maintain/concerns/primary_checks.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            raise StandardError, "Error: Important rpm package #{package} is not installed!"\
+                  "\nInstall #{package} rpm to ensure system consistency."
+          end
+        end
+      end
+
+      def package_name
+        installed_scenario = detector.feature(:installer).last_scenario
+        if installed_scenario == 'satellite'
+          'satellite'
```
Comment (kgaikwad) [inline lib/foreman_maintain/concerns/primary_checks.rb:17]: Features `capsule` & `satellite` have `package_name` method but those are not declared as class methods. 
If those will be class methods then I guess we can access them here and no need to hardcode them at multiple places.
BTW, this is just a thought so I don't want to block the PR due this change.

**Review by kgaikwad — CHANGES_REQUESTED** [review]
@upadhyeammit, overall looks good to me :+1: 
Added two inline suggestions.

File: lib/foreman_maintain/concerns/primary_checks.rb (hunk 1 above)
Comment (upadhyeammit) [reply lib/foreman_maintain/concerns/primary_checks.rb]: Modified to use class methods.

File: lib/foreman_maintain/concerns/primary_checks.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module ForemanMaintain
+  module Concerns
+    module PrimaryChecks
+      def validate_downstream_packages
+        package = package_name
+        if !package.nil? && detector.feature(:installer).with_scenarios?
+          unless package_manager.installed?(package)
+            raise StandardError, "Error: Important rpm package #{package} is not installed!"\
+                  "\nInstall #{package} rpm to ensure system consistency."
```
[the remark this answers is not in the document]
Comment (upadhyeammit) [reply lib/foreman_maintain/concerns/primary_checks.rb]: Updated to use "ForemanMaintain::Error::Fail"

**Review by kgaikwad — APPROVED** [review]
@upadhyeammit, Not tested but changes look good to me!

**Comment by upadhyeammit** [comment]
@jameerpathan111 request to test

**Comment by upadhyeammit** [comment]
@kgaikwad as per the offline discussion we are going with not changing package_name methods of satellite and capsule features to class methods. I will raise separate pr for same with more testing coverage on same. Thank You for reviews!

**Pull request merged**